Src/CSharpier/SyntaxPrinter/Modifiers.cs

```diff
@@ -2,34 +2,92 @@ namespace CSharpier.SyntaxPrinter;
 
 internal static class Modifiers
 {
-    public static Doc Print(SyntaxTokenList modifiers, FormattingContext context)
+    private class DefaultOrder : IComparer<string>
     {
-        if (modifiers.Count == 0)
+        // use the default order from https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0036
+        private static readonly string[] DefaultOrdered =
         {
-            return Doc.Null;
+            "public",
+            "private",
+            "protected",
+            "internal",
+            "file",
+            "static",
+            "extern",
+            "new",
+            "virtual",
+            "abstract",
+            "sealed",
+            "override",
+            "readonly",
+            "unsafe",
+            "required",
+            "volatile",
+            "async",
+        };
+
+        public int Compare(string? x, string? y)
+        {
+            int GetIndex(string? value)
+            {
+                var result = Array.IndexOf(DefaultOrdered, value);
```

Comment:
So, this gets called for every comparison of _every_ declaration in the source code, right?
It might be worth optimizing to be constant time access by say, initializing a static dictionary of string to index.